### DISCUSSION TITLE
Example in shortest_path()

### DISCUSSION THREAD

**Discussion by NikitaSharma1** [networkx/networkx#5483] [Q&A] [answered]
I think the example given in the document of shortest_path() is not clearly showing the difference between the different types of output one will get on specifying source, target, both or none. It will be helpful to show the different outputs in the example. 
If that's okay I have already worked on it but I am not able to find the rst file, can someone please tell me where can I find it?

**Comment by MridulS** [comment, marked as answer]
If we are talking about https://networkx.org/documentation/latest/reference/algorithms/generated/networkx.algorithms.shortest_paths.generic.shortest_path.html#networkx.algorithms.shortest_paths.generic.shortest_path then yes the variable `p` can be confusing, but printing everything will make the example a bit messier as it will print a bunch of dictionaries. Could you maybe add a comment about what the return dictionary `p` is?
Something like
``` python
>>> p = nx.shortest_path(G, source=0)  # target not specified
>>> p[4] # shortest path to node 4 from source 0
[0, 1, 2, 3, 4]
>>> p[3] # shortest path to node 3 from source 0
[0, 1, 2, 3]
```
To edit this documentation bits you can edit the function docstring for `shortest_path` directly, the documentation generation pipeline will parse these examples and create documentation. No need to edit any rst file :)
https://github.com/networkx/networkx/blob/2348127e8b874b98990b207d0473ca6ebc3b5a2f/networkx/algorithms/shortest_paths/generic.py#L98-L110
